enterprise/frontend/src/embedding-sdk/lib/theme/get-embedding-theme.ts

```diff
@@ -10,12 +17,17 @@ import { colorTuple } from "./color-tuple";
 export function getEmbeddingThemeOverride(
   theme: MetabaseTheme,
 ): EmbeddingThemeOverride {
+  const components: MetabaseComponentTheme = merge(
```

Comment:
This shallow merges the 2 objects. Is that what you intended? Asking because both object have nested properties so any nested properties won't be merged.

Reply:
@WiNloSt In the icepick docs it says it performs a [deep merge](https://github.com/aearly/icepick?tab=readme-ov-file#mergetarget-source-associator). Seems to be merging fine from my testing.

> Deeply merge a source object into target, similar to Lodash.merge. Child collections that are both frozen and reference equal will be assumed to be deeply equal. Arrays from the source object will completely replace those in the target object if the two differ. If nothing changed, the original reference will not change. Returns a frozen object, and works with both unfrozen and frozen objects.

Reply:
My bad, I was testing this wrong. You're right. it correctly deep merges the 2 objects. Sorry for the confusion.